src/xb-database-manager.c

```diff
@@ -658,8 +650,26 @@ xb_database_manager_query (XbDatabaseManager *self,
       goto out;
     }
 
-  parsed_query = xapian_query_parser_parse_query_full (payload->qp, query_str,
-                                                       QUERY_PARSER_FLAGS, "", &error);
+  const gchar *match_all;
+  match_all = g_hash_table_lookup (query_options, QUERY_PARAM_MATCH_ALL);
+  if (match_all != NULL && str == NULL)
```

Comment:
Not sure what's the best strategy here; if both are given then ignore the query string? Or throw the error?